projeto_guiado_v2_williane.py

```diff
@@ -0,0 +1,187 @@
+            consultar_faltas_aluna()
+        elif cod_opcao == 4:
+            consultar_notas_aluna()
+        elif cod_opcao == 5:
+            consultar_status_aprovacao()
+        elif cod_opcao == 6:
+            print("Encerrando o programa...")
+            break
+
+def obter_opcao():
+    while True:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+            if codigo_opcao in [1, 2, 3, 4, 5, 6]:
+                return codigo_opcao
+            else:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 6).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+
+def incluir_nova_aluna():
+    nome = input("\nDigite o nome completo da aluna: ")
+    sobrenome = input("Digite o sobrenome da aluna: ")
+    turma = input("Digite a turma da aluna: ")
+    
+    # Coletando as notas
+    quantidade_notas = int(input("Quantas notas serão inseridas? "))
+    notas_aluna = []
+    for i in range(quantidade_notas):
+        while True:
+            try:
+                nota = float(input(f"Insira a nota #{i+1}: "))
+                if nota < 0 or nota > 10:
+                    print("Nota fora do intervalo permitido (0-10). Tente novamente.")
+                else:
+                    notas_aluna.append(nota)
+                    break
+            except ValueError:
+                print("Valor inválido. Insira um número válido para a nota.")
+    
+    # Coletando a presença (faltas)
+    quantidade_dias = int(input("Quantos dias de presença serão inseridos? "))
+    presenca_aluna = []
+    for i in range(quantidade_dias):
+        while True:
+            try:
+                presenca = input(f"Informar presença no dia #{i+1} (True/False): ").strip().lower()
+                if presenca not in ['true', 'false']:
+                    raise ValueError("Valor inválido. Insira 'True' ou 'False'.")
+                else:
+                    presenca_aluna.append(presenca == 'true')
+                    break
+            except ValueError as e:
+                print(e)
+    
+    # Coletando a nota de participação
+    while True:
+        try:
+            participacao = float(input("Insira a nota de participação da aluna: "))
+            if participacao < 0 or participacao > 10:
+                raise ValueError("Nota de participação fora do intervalo permitido (0-10).")
+            else:
+                break
+        except ValueError as e:
+            print(e)
```

Comment:
Aqui não necessáriamente precisaria usar um raise ValueError, poderia ser um print. Assim, teriamos a mesma lógica com implementação mais simples. Neste ponto, caso insiramos uma letra, recebemos uma mensagem de exception não tratado, que é a mensagem retornada pela linha 87.

**Sugestão:**
```suggestion
            if participacao < 0 or participacao > 10:
                print("Nota de participação fora do intervalo permitido (0-10).")
            else:
                break
        except ValueError:
            print("Entrada inválida. Por favor, insira um número.")
```